apps/whale-api/src/module.api/poolswap.pathfinding.service.ts

```diff
@@ -255,18 +268,45 @@ export class PoolSwapPathFindingService {
   }
 }
 
-function computeReturnInDestinationToken (path: SwapPathPoolPair[], fromTokenId: string): BigNumber {
-  let total = new BigNumber(1)
+function computeReturnLessDexFeesInDestinationToken (path: SwapPathPoolPair[], fromTokenId: string): {
+  estimatedReturn: BigNumber
+  estimatedReturnLessDexFees: BigNumber
+} {
+  let estimatedReturnLessDexFees = new BigNumber(1)
+  let estimatedReturn = new BigNumber(1)
+  let priceRatio, fromTokenFeePct, toTokenFeePct
+
   for (const poolPair of path) {
     if (fromTokenId === poolPair.tokenA.id) {
-      total = total.multipliedBy(poolPair.priceRatio.ba)
       fromTokenId = poolPair.tokenB.id
+      priceRatio = poolPair.priceRatio.ba
+      fromTokenFeePct = poolPair.estimatedDexFeesInPct?.ba
+      toTokenFeePct = poolPair.estimatedDexFeesInPct?.ab
     } else {
-      total = total.multipliedBy(poolPair.priceRatio.ab)
       fromTokenId = poolPair.tokenA.id
+      priceRatio = poolPair.priceRatio.ab
+      fromTokenFeePct = poolPair.estimatedDexFeesInPct?.ab
+      toTokenFeePct = poolPair.estimatedDexFeesInPct?.ba
     }
+
+    estimatedReturn = estimatedReturn.multipliedBy(priceRatio)
+
+    // less dex fee token A
+    const estimatedDexFeeInTokenA = new BigNumber(fromTokenFeePct ?? 0).multipliedBy(estimatedReturnLessDexFees)
```

Comment:
Correct me if I'm wrong, but I think `estimatedDexFeeInTokenA` here is misleading as `fromToken` could refer to either token A or B (not just A) right?

Reply:
You're right, it's quite misleading and better to use fromToken, updated!